src/xcode/ENA/ENA/Source/Views/BaseElements/ENATanInput.swift

```diff
@@ -346,7 +346,7 @@ private class ENATanInputLabel: UILabel {
 
 		self.textColor = textColor
 
-		if false != text?.isEmpty {
+		if false != text?.isEmpty || !isValid {
```

Comment:
this does the job but is hard to read. is there a way to make this nicer? if there is no way, maybe add a comment?

Reply:
```suggestion
		if text?.isEmpty ?? true || !isValid {
```
this assumes that the reader understands `Optionals` in general and `??` specifically.
```suggestion
		if nil == text || text!.isEmpty || !isValid {
```
this assumes that the reader (including code quality QA tools) is aware of _short circuit evaluation_, as without `nil == text ||` the `text!.isEmpty ` would potentially result in an runtime error. `!.isEmpty` could be read as "not empty" by a non-swift programmer. To prevent that, brackets could be used:
```suggestion
		if nil == text || (text!).isEmpty || !isValid {
```
I am not a swift programmer. For me, my third suggestion is easiest to read, as I only would have to look up what the `!` in `(text!)` means.

Reply:
Hi @EisvogelSAP ,
Would you mind adding some comment in your code which helps the upcoming developers to maintain it.
Thanks.